internal/explorer/helpers.go

```diff
@@ -287,21 +290,58 @@ func (a *App) getNodeData(nodeID string, force bool) (string, error) {
 			}
 			return "", ErrNodeNotFound
 		} else if err != nil {
-			err = a.DeleteRedisKey(fmt.Sprintf("GRID3NODE:%s", nodeID))
-			if err != nil {
-				log.Warn().Err(err).Msg("could not delete key in redis")
+
+			redisData := NodeInfo{}
+			json.Unmarshal([]byte(value), &redisData)
+
+			// up or loading make it likely down for 10 minutes
+			if redisData.Status == "up" {
+				// mark the node likely down if we can't reach this node in 10 mins it's down
+				nodeInfo.Status = "likely down"
+				marshalledInfo, err := json.Marshal(nodeInfo)
+				if err != nil {
+					log.Error().Err(err).Msg("could not marshal node info")
+					return "", errors.Wrap(err, "internal server error")
+				}
+				err = a.SetRedisKey(fmt.Sprintf("GRID3NODE:%s", nodeID), marshalledInfo, 10*60)
+				if err != nil {
+					log.Warn().Err(err).Msg("could not cache data in redis")
+				}
+
+				return "", ErrBadGateway
+			}
+
+			// down will be still down
+			if redisData.Status == "down" || redisData.Status == "" {
+				if err != nil {
+					log.Warn().Err(err).Msg(fmt.Sprintf("node: %s will be marked as down", nodeID))
+				}
+				nodeInfo.Status = "down"
+				marshalledInfo, err := json.Marshal(nodeInfo)
+				if err != nil {
+					log.Error().Err(err).Msg("could not marshal node info")
+					return "", errors.Wrap(err, "internal server error")
+				}
+				err = a.SetRedisKey(fmt.Sprintf("GRID3NODE:%s", nodeID), marshalledInfo, 30*60)
+				if err != nil {
+					log.Warn().Err(err).Msg("could not cache data in redis")
+				}
+				return "", ErrBadGateway
 			}
+
+			log.Warn().Err(err).Msg(fmt.Sprintf("node %s is likely down", nodeID))
 			return "", ErrBadGateway
```

Comment:
Shouldn't this has the actual error that was returned by querying the node info. BadGateway is so generic and doesn't tell us much about the underlying error. You still can wrap the errors so it's a BadGateway but with details about what caused it.